package/piwebcam/multi-gadget.sh

```diff
@@ -86,14 +86,27 @@ config_usb_webcam () {
 }
 
 # Check if camera is installed correctly
-if [ ! -e /dev/video0 ] ; then
-  echo "I did not detect a camera connected to the Pi. Please check your hardware."
-  CONFIGURE_USB_WEBCAM=false
+if [ -e /dev/video0 ] ; then
+  CONFIGURE_USB_WEBCAM=true
+else
+  COUNT=0
+  while [ $COUNT -lt 3 ] ; do
+      if [ -e /dev/video0 ] ; then
+        CONFIGURE_USB_WEBCAM=true
+        echo "Camera detected after $COUNT seconds."
+        break
+      fi
+      sleep 1
+      COUNT=$((COUNT+1))
+  done
 fi
 
 if [ "$CONFIGURE_USB_WEBCAM" = true ] ; then
   echo "Configuring USB gadget webcam interface"
   config_usb_webcam
+else
+  echo "No camera detected after 3 seconds. Enabled serial for debugging."
+  CONFIGURE_USB_SERIAL=true
```

Comment:
That's a good idea though - if the camera doesn't load, it should force the serial to be enabled I think.

Reply:
There's a good chance that serial won't work, as uvc-gadget fails. Didn't test this for this specific case though.

Reply:
It could be useful at least for debugging camera issues. If you don't have uvc then you have bigger issues. Say your camera cable came loose or your camera didn't register in time this can help debug were the problem is.

Reply:
> There's a good chance that serial won't work, as uvc-gadget fails. Didn't test this for this specific case though.

No, it's fine. If the configfs isn't set up (because config_usb_webcam isn't called) then there will be nothing blocking in the composite gadget, and it won't matter if uvc-gadget isn't running.

Reply:
@dgsiegel I just tested with the latest version of commits here (99bf203) and you were right, with no camera attached the device doesn't register properly.

Even when it does appear as a serial gadget it's been near unusable for me, constantly freezing up often before I can even log in. Is it worth switching to something like rndis network gadget instead? Although that does come with a little more overhead.

Then again, it may well just be some missing config option.

Reply:
That's interesting, the serial gadget worked well for me when I used it this week.
But the freezing/lockups sound similar to what I was experiencing while testing gadget (outside of show-me-webcam)

Reply:
I'll have to do some testing... rndis has been a much more reliable experience than USB_G_SERIAL for me in the past. I've just made the script enable serial if the camera isn't detected again, so I'll mark this as resolved.